### DISCUSSION TITLE
I can only search within the feeds

### DISCUSSION THREAD

**Discussion by sassanix** [FreshRSS/FreshRSS#6490] [Q&A] [answered]
Hi everyone,

I can only search the feeds, but I can't search my mainstream or categories. How can I fix this?

**Comment by Alkarex** [comment]
Hello, 
You need to provide more information, including about your system. A screenshot would help, too

**Reply by sassanix** [reply]
I'm running the latest version of FreshRSS, running on Docker.

**Reply by Alkarex** [reply]
And what are the symptoms? A screenshot would also help

**Reply by sassanix** [reply]
![image](https://github.com/FreshRSS/FreshRSS/assets/39465071/7cd82631-8fd4-43ed-8560-989a03378219)

Does this help? This is what I get when I search for anything within categories, and within mainstream. But it works within individual feeds.

**Reply by Alkarex** [reply]
Right, it might be because you haven't any unread article. Try to click the other envelope as well, to display read articles

**Reply by sassanix** [reply]
That works for my read articles. But unread articles, I don't see anything. Does it only search articles I've read?

**Reply by Alkarex** [reply]
With both envelopes activated, you will search in both read and unread articles

**Reply by sassanix** [reply]
For example here 
![image](https://github.com/FreshRSS/FreshRSS/assets/39465071/f62e3dd9-f5dd-4e07-a908-81592fe1f6a0)

I have both unread and read selected. 

Then I search for 
![image](https://github.com/FreshRSS/FreshRSS/assets/39465071/54aee5c2-518c-43aa-8ec5-0b870755af69)


I don't see those articles showing.

**Reply by Alkarex** [reply]
Hum, that is strange. Have you changed the *Visibility* setting of the feeds? Make sure they are set to *Show in main stream*.

![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/1ab91abd-2fdb-4fe7-84bf-fc34433256d2)

What database are you using? And what FreshRSS Docker image, more precisely?

**Reply by sassanix** [reply]
Hi Alkarex, I have it visible and shown in the main stream.

The docker repo is maintained by: lscr.io/linuxserver/freshrss

For my database I'm using mariadb.

I have done all the latest updates to Freshrss and Mariadb.

**Reply by Alkarex** [reply]
Can you reproduce the same behaviour on e.g. https://demo.freshrss.org ?
`linuxserver/freshrss` is not our official Docker image (which is `freshrss/freshrss`), but it should probably also work

**Reply by sassanix** [reply]
> Can you reproduce the same behaviour on e.g. https://demo.freshrss.org ? `linuxserver/freshrss` is not our official Docker image (which is `freshrss/freshrss`), but it should probably also work

I can't replicate it on there, search works as intended on the demo page.

**Comment by math-GH** [comment]
Could you please double check:
- in "about" section: is it FreshRSS `1.24.0`?
- which extensions are enabled?
- in "log" section: is the log empty?
- if possible: any other logs (f.e. the PHP php_error_log) checked? Browser console is empty too?
- search for `intitle:Taiwan`. Is there any result?
- could you check please another 2-3 keywords?

**Reply by sassanix** [reply]
1. About page: It's 1.24.0
2. ReadingTime is the only extension enabled
3. No it's not empty, I have this error a lot:  SQL error Fresh	ReadingTimeRSS_FeedDAO::updateCachedValues["HY000",1034,"Index for table 'Sassan_entry' is corrupt; try to repair it"]   When I tried to repair it within phpmyadmin, it wouldn't as it says the engine doesn't support it.
4. I will check to see how to find these logs.
5. No results.
6. Checked 4 different keywords and no results.

**Reply by Alkarex** [reply]
If your database appears corrupted, I suggest (after your make a backup of your system) to attempt a backup & restore with `cli/db-backup.php` and `cli/db-restore.php`
https://freshrss.github.io/FreshRSS/en/admins/05_Backup.html#creating-a-database-backup
For Docker, you can prefix the commands with `docker exec freshrss cli/...`

**Reply by sassanix** [reply]
I will work on that and let you know if it all works out.

**Comment by sassanix** [comment, marked as answer]
I finally managed to resolve the issue. Here’s what I did:

First, I accessed the server’s shell with:

```bash
docker exec freshrss /bin/bash
```

Then, I navigated to the CLI directory:

```bash
cd /app/www/cli
```

There, I ran the backup and restore scripts:

```bash
php db-backup.php
php db-restore.php --force-overwrite
```

Unfortunately, I lost most of my favourites during the process, but my subscriptions remained intact. Interestingly, I noticed that my non-English subscriptions were lumped under "uncategorized," and one didn't even display a title. It seemed that might have been the root of the problem. I re-entered the feed information for the problematic subscription, then deleted and reorganized the others from the uncategorized section.

After all that, I still couldn’t get the feed to update. So, I decided to connect to the MariaDB database and restarted the container.

Now, my feeds are refreshing correctly, and articles are coming through again. It's disappointing to have lost some intriguing articles, but on the bright side, everything's functioning properly now. I double-checked by testing the homepage and the search feature—it's now effectively searching across all feeds, which is fantastic!

**Reply by Alkarex** [reply]
It looks like some symptoms of a corrupted database.
We offer multiple export features (OPML, XML of the favourites / labels, SQLite of everything), which can also be used as backups, and which should be run regularly.

**Reply by sassanix** [reply]
I did make a backup, but when I tried to import my starred json file, it gave me an error "File cannot be uploaded!"

**In the logs, it says:** File cannot be uploaded. Error code: 1

**Reply by Alkarex** [reply]
We can try to recover your backup. 
If there is nothing too private in it, you are welcome to send the file to me. 
Otherwise, try first to open it in an editor and XML viewer

**Reply by sassanix** [reply]
There's nothing personal there, just some articles I've meant to read.

As part of my troubleshooting, I used XML Notepad to view the contents, saved it again, but unfortunately, I encountered the same error in FreshRSS.

I then tried to just get the URL's through a python script, which worked. I can now see the URL's better, and go through them and bookmark the ones that I want. 

But it would be nice to figure out how to repair the JSON file.

**Reply by Alkarex** [reply]
Just [send me](https://alexandre.alapetite.fr/cv/) your JSON file then and I will try to see how to fix it.

**Reply by sassanix** [reply]
Sent you the file.

**Reply by Alkarex** [reply]
I tried to import your file, and it worked without problem. Make sure you check the favourites section (the number should have increased to e.g. 245) and show also the read articles:
![image](https://github.com/FreshRSS/FreshRSS/assets/1008324/60625625-9c40-40c1-8ccd-17577dd20228)

If it does not work on your side, try to look at your Web server logs for some errors / warnings

**Reply by sassanix** [reply]
I tried it again, and I got the error that it can't be uploaded again.

This is what I see in the logs:

![image](https://github.com/FreshRSS/FreshRSS/assets/39465071/df378be9-44c7-4892-b1c8-6d8e0f3051af)

**Reply by Alkarex** [reply]
It looks like your `php.ini` settings for uploading files, `MAX_FILE_SIZE` in particular, are too small
https://php.net/features.file-upload.errors

**Reply by Alkarex** [reply]
And those are the FreshRSS logs. Check your Web server logs as well

**Reply by sassanix** [reply]
Thank you so much!

I edited the php-local.ini to include:

```

upload_max_filesize = 100M
post_max_size = 100M
```

Then I restarted the server.

And the upload worked, I've now got all of my previous favourites in my freshrss, and the search is working as well!

**Reply by sassanix** [reply]
I do notice that I have duplicate feeds in uncategorized now though, if I remove the feed from uncategorized it will remove the article that I have in the favourites.

**Reply by Alkarex** [reply]
> I have duplicate feeds in uncategorized now though

That is something to improve. Please open a new issue

**Reply by Alkarex** [reply]
Regarding your duplicated feeds, do they have the same feed URL?
If they do not have the exact same feed URL, they will not be considered as the same feed

**Reply by sassanix** [reply]
They have the same feed URL, I use five filters so they're all the same URL but for different feeds.

Should I open up an issue for that?

**Reply by Alkarex** [reply]
Yes, please open an issue with more details and screenshots and/or OPML export of the feed details

**Reply by sassanix** [reply]
> Yes, please open an issue with more details and screenshots and/or OPML export of the feed details

I'll make sure to do that, I'm happy that I can search all of my feeds again! Thanks a lot for your help.
